src/SQLProvider.Runtime/Providers.MsSqlServer.Ssdt.fs

```diff
@@ -165,32 +165,31 @@ module MSSqlServerSsdt =
     let tryFindMapping (dataType: string) =
         typeMappingsByName.TryFind (dataType.ToUpper())
 
-    let tryFindMappingOrVariant (dataType: string) =
-        match typeMappingsByName.TryFind (dataType.ToUpper()) with
+    let rec tryFindMappingOrVariant (uddts: SsdtUserDefinedDataType) (dataType: string) =
+        let dataType = dataType.ToUpper()
+        match typeMappingsByName.TryFind dataType with
         | Some tm -> tm
-        | None -> (typeMappingsByName.TryFind "SQL_VARIANT").Value
+        | None ->
+            match Map.tryFind (UDDTName dataType) uddts with
+            | Some (UDDTInheritedType x) -> tryFindMappingOrVariant uddts x
+            | None -> (typeMappingsByName.TryFind "SQL_VARIANT").Value
 
     let ssdtTableToTable (tbl: SsdtTable) =
         { Schema = tbl.Schema ; Name = tbl.Name ; Type =  if tbl.IsView then "view" else "base table" }
 
-    let ssdtColumnToColumn (tbl: SsdtTable) (col: SsdtColumn) =
```

Comment:
Hey @Thorium, about this function, I was doubtful if I should remove the Option from it since I am unaware of the consequences it might lead into in the `GetColumns` member with the List.choose. In any case, I removed it in favor of the UDDT map

Reply:
I'm not familiar with this, @JordanMarr knows better. Maybe the idea was that totally unknown column types are just skipped by SQLProvider. I would expect that e.g. stored procedures could return something unknown that is then just treated as an obj.

Reply:
It was an arbitrary decision on my part to omit the column instead of adding as a variant / `obj`.
Your change to call `tryFindMappingOrVariant` seems ok to me.